examples/qml_features/rust/src/custom_base_class.rs

```diff
@@ -53,6 +53,25 @@ mod ffi {
         fn end_reset_model(self: Pin<&mut CustomBaseClassQt>);
     }
 
+    // TODO: have a way to generate the type specific code
+    //
+    // TODO: can this also define that QSetI32 is a type alias?
+    //
+    // impl cxx_qt_lib::QSet<i32> {}
+    //
+    // TODO: generate the following
+    unsafe extern "C++" {
+        // TODO: the include won't be required if the C++ is generated
+        include!("cxx-qt-lib/qset.h");
+        type QSetI32 = super::private::QSetI32;
```

Comment:
Hm, we could generate this part from:
``` rust
type QSetI32 = cxx_qt_lib::QSet<i32>;
```
As well as the typedef `using QSetI32 = QSet<i32>;` in the C++ side.

However, then we'd have to again start scanning `extern "C++"` blocks, which we want to avoid...

Reply:
Hm, it should also be doable to support a templated extern type :thinking: 
https://github.com/dtolnay/cxx/issues/251

And more specifically:
https://github.com/dtolnay/cxx/pull/524/files#diff-a9b6bb47cc68e30ec36cb376284aaa6fdee44f9fc71308fc043df611e760f7a1

Reply:
Right this works :-)